server-code/Projects.js

```diff
@@ -2,12 +2,52 @@
 Document: JS code specific for projects
 */
 
+// constants for 'Slide Deck' and 'BOM' text, to follow best practice of keeping things in one place
+const SLIDE_DECK_STR = "Slide Deck";
+const BOM_STR = "BOM";
+
 /**
  * Returns HTML formatted list of all projects.
  * 
  * @return {String} – A constructed HTML table listing all the projects
  */
 function getAllProjects() {
     var data = getSheetInfo('mainSheetID', 'Projects', 'data');
+    transformToHyperLinks(data);
     return buildTableHTML(data, "table-sm");
 }
+
+/**
+ * Creates an HTML hyperlink, given the display text, that requests the given url.
+ * 
+ * @param {String} url – The url to link to
+ * @param {String} displayText – The text that will be displayed
+ * @return {String} – The corresponding HTML hyperlink
+ */
+function getHTMLLink(url, displayText) {
+    var html = `<a href="` + url + `">` + displayText + `</a` 
+    return html
+}
+
+/**
+ * Transforms the links in the Slide Deck and BOM columns into HTML links with the display text of 
+ * "Slide Deck" or "BOM". 
+ * 
+ * @param {Object[][]} data – Spreadsheet data from the Projects table in the database
+ * @return {void}
+ */
+function transformToHyperLinks(data) {
+    var headers = data[0];
+    var slideDeckColIdx = findIdx(SLIDE_DECK_STR, headers);
+    var bomColIdx = findIdx(BOM_STR, headers);
+    for (var rowIdx = 1; rowIdx < data.length; rowIdx++) {
+        slideDeckURL = data[rowIdx][slideDeckColIdx];
+        bomURL = data[rowIdx][bomColIdx];
+        if (slideDeckURL && bomURL) {
```

Comment:
this is good! quick thought: what happens when bomURL is populated but slideDeckURL is an empty string? are they ever null values? I think the answer is no, but I could be wrong. Would it be better to do the if as two inline if statements sorta like:
 `slideDeckURL.length > 0 ? getHTMLLink(slideDeckURL, SLIDE_DECK_STR) : "" // converts url to HTML or leaves as empty string`

Reply:
Agreed. This implementation would only work if both URLs are populated. Combining your method of checking for null/empty strings and James' inline statement, you could also use something like:
`slideDeckURL ? getHTMLLink(slideDeckURL, SLIDE_DECK_STR) : "" // converts url to HTML or leaves as empty string`

Reply:
Yeah I had just assumed that probably there would not be such a case but I guess it’s possible that may not always be the case 
Will change